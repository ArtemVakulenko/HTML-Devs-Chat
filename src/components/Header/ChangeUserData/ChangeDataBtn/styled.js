import styled from 'styled-components';

export const StButtonUser = styled.div`
  display: flex;
  flex-direction: column;
  margin: 0 15px;
  button{
    margin: 5px 0 0 0;
  }
  span{
    font-size: 12px;
  }
`;
