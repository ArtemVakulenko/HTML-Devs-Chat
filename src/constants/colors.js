export const colors = {
  dark: {
    main: '#0E1621',
    secondary: '#17212B',
    input: '#242F3D',
    header: '#2B5278',
    button: '#2B5278',
    message: '#2B5278',
    buttonText: '#fff',
    notification: '#5C9DDD',
    textError: '#ff0000',
    color: '#fff',
    font: '#ffffff',
    placeholder: '#55667B',
    modal: '#0E1621',
    modalWrapper: 'rgba(0, 0, 0, 0.5)',
    selectColor: '#000',
    guestName: '#488ccf',
  },
  light: {
    main: '#FFFFFF',
    secondary: '#ECECEC',
    input: '#D1D1D1',
    header: '#2B5278',
    button: '#2B5278',
    message: '#2B5278',
    buttonText: '#fff',
    notification: '#5C9DDD',
    textError: '#ff0000',
    color: '#000',
    font: '#ffffff',
    placeholder: '#55667B',
    modal: '#FFFFFF',
    modalWrapper: 'rgba(0, 0, 0, 0.5)',
    selectColor: '#000',
    guestName: '#2B5278',
  },
};
