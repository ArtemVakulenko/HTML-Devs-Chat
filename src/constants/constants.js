export const langs = {
  arabian: 'ar',
  russian: 'ru',
  english: 'en',
};

export const cookieAge = 20000;
