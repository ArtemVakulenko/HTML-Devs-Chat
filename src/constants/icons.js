export const icons = {
    changeThemeIcon: '../assets/img/changeThemeIcon.png',
    defaultUserIcon: '../assets/img/defaultUserIcon.png',
    dropDownIcon: '../assets/img/dropDownIcon.png',
    logoIcon: '../assets/img/logoIcon.png',
    logOutIcon: '../assets/img/logOutIcon.png',
    settingIcon: '../assets/img/settingsIcon.png',
    closeModalDark: '../assets/img/closeModalDark.png',
    sendMessage: '../assets/img/sendMessage.png',
    showPass: '../assets/img/view.png',
    hidePass: '../assets/img/hide.png',
    themeDark: '../assets/img/moon.png',
    themeLight: '../assets/img/sun.png',
};
