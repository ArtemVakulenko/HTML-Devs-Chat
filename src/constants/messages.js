export const MESSAGES_KEYS = {
  invalidLogin: 'invalidLogin',
  invalidPassword: 'invalidPassword',
  invalidConfirm: 'invalidConfirm',
  userIsReg: 'userIsReg',
  notExist: 'notExist',
  wrongPass: 'wrongPass',
  abstract: 'something went wrong',
  successReg: 'successReg',
  successAuth: 'successAuth',
  unauth: 'unAuth',
};
