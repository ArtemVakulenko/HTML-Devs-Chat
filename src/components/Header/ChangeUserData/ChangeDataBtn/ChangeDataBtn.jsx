import React from 'react';
import PropTypes from 'prop-types';
import { icons } from 'constants';
import Button from 'src/components/_common/Button';
import { StButtonUser } from './styled';

const ChangeDataBtn = ({ data, onClick }) => {
  return (
    <StButtonUser >
      <Button
        icon={data.img || icons.defaultUserIcon}
        onClick={onClick}
      />
      <span>{data.login}</span>
    </StButtonUser>
  );
};

export default ChangeDataBtn;

ChangeDataBtn.propTypes = {
  data: PropTypes.object,
  onClick: PropTypes.func.isRequired,
};
