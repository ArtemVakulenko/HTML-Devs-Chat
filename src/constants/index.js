import { colors } from './colors';
import { icons } from './icons';
import { MESSAGES_KEYS } from './messages';
import { url } from './url';
import { LOGIN, PASSWORD } from './regExp';
import { langs, cookieAge } from './constants';

export { url, colors, icons, MESSAGES_KEYS, LOGIN, PASSWORD, langs, cookieAge };
