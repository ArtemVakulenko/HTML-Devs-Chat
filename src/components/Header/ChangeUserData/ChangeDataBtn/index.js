import { connect } from 'react-redux';
import { changeButtonData } from 'store/mainPage/selectors';
import ChangeDataBtn from './ChangeDataBtn';

const mapStateToProps = (state) => ({
  data: changeButtonData(state),
});
export default connect(mapStateToProps)(ChangeDataBtn);
