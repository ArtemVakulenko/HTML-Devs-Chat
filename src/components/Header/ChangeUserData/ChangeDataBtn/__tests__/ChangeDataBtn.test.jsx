import { shallow, mount } from 'enzyme';
import React from 'react';
import { icons } from '../../../../../constants';
import ChangeDataBtn from '../ChangeDataBtn';

jest.mock('react-i18next', () => ({
  useTranslation: jest.fn().mockImplementation(() => ({
    t: jest.fn(),
  })),
}));

describe('ChangeDataBtn', () => {
  const props = {
    data: {
      login: 'login',
      img: 'img',
    },
    onClick: jest.fn(),
  };
  it('Should match snapshot', () => {
    const component = shallow(<ChangeDataBtn {...props} />);
    expect(component.html()).toMatchSnapshot();
  });
  it('Should have Button', () => {
    const component = mount(<ChangeDataBtn {...props} />);
    expect(component.find('Button')).toHaveLength(1);
  });
  it('Should have StButtonUser', () => {
    const component = mount(<ChangeDataBtn {...props} />);
    expect(component.find('styled__StButtonUser')).toHaveLength(1);
  });
  it('Should have span', () => {
    const component = mount(<ChangeDataBtn {...props} />);
    expect(component.find('span')).toHaveLength(1);
  });
  it('Should have Button have img', () => {
    const component = mount(<ChangeDataBtn {...props} />);
    expect(component.find('Button').props().icon).toBe(props.data.img);
  });
  it('Should have Button have default img', () => {
    props.data.img = '';
    const component = mount(<ChangeDataBtn {...props} />);
    expect(component.find('Button').props().icon).toBe(icons.defaultUserIcon);
  });
});
