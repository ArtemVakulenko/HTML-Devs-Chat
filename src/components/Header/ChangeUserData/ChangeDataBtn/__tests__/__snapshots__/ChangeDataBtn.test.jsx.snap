// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangeDataBtn Should match snapshot 1`] = `"<div class=\\"styled__StButtonUser-bYtjyH gNhnyN\\"><button class=\\"styled__StIcon-chPeWD eZnRst\\"></button><span>login</span></div>"`;
